src/main/java/seedu/address/model/person/TagEqualsPredicate.java

```diff
@@ -0,0 +1,35 @@
+package seedu.address.model.person;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Person} has a tag that matches the given tag name.
+ */
+public class TagEqualsPredicate implements Predicate<Person> {
+    private final String tagName;
+
+    public TagEqualsPredicate(String tagName) {
+        this.tagName = tagName;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return person.getTags().stream()
+                .anyMatch(tag -> tag.tagName.equals(tagName));
+    }
```

Comment:
Maybe can use .map(x -> x.toLowerCase()) to set all tags to lowercase to compare their value.

Reply:
Thanks for your feedback! I have modified the code with feedback that you suggested.